### DISCUSSION TITLE
右側トラックボール有りの２キー程反応がありません

### DISCUSSION THREAD

**Discussion by JinBallMen** [Yowkees/keyball#283] [Q&A]
お世話になっております。

右側トラックボールの以下２箇所（一番右側の列の２キー）について、キーボード入力のチェックを行いましたが反応がありませんでした。どういった原因が考えられますでしょうか。

![20230723_130548~2](https://github.com/Yowkees/keyball/assets/75989686/6c847ed1-c41c-4d54-aa7a-521183f08033)

キーソケットは問題なさそうでした。
また、ダイオードについては、２箇所のキーのすぐ上のダイオードを修正しても変化なしでした（もし別の位置と繋がっているのであればご教示ください）。
一番怪しいのはProMicro側かと思っておりますが、どこを確認すれば良いかわからず一度ProMicro側のはんだだけつけ直しております。

以下試したことです。
・キーソケットのはんだつけ直し（つけ直す前も、浮いているようには見えず、あまり関係ない？）
・周辺のダイオードのはんだつけ直しと位置直し（一部傾いていたため修正、位置を直しても変化なし）
・テスターで導通の確認
　・ダイオード側　→　抵抗値が表示される
　・キーソケット押下での確認　→　ピッと鳴り抵抗値が表示される
・ProMicro側の問題？と考え、はんだつけ直し（付属の12ピンヘッダが破損してしまっているので、ProMicroに付属のもので代替していますが、それが問題？）

以上、お忙しい中大変申し訳ありませんが
よろしくお願いいたします。

**Comment by Yowkees** [comment]
ご連絡ありがとうございます。

半田付けを見直していただいた２か所のダイオードのほかに、下記画像の緑丸をつけたダイオードも、問題の２キーに関わっていますので、テンプラしていないかご確認ください。

また、ProMicroの半田付けが原因の場合、下記画像の緑点２点の接続不良や、隣とのショートがあるとキー入力不具合となりそうですので、こちらも見直してみてください。

以上、お手数ですがご確認お願いいたします。

![KB61R_2keyfix](https://github.com/Yowkees/keyball/assets/84209772/febab94f-e278-4005-826c-f5f1be513e3b)

**Comment by JinBallMen** [comment]
回答いただきありがとうございます。拡大鏡を使い全体を見直してみたところ、もっと初歩的なミスでした。
ダイオードを反対に実装している箇所を修正し、テンプラしないよう慎重にはんだを充ててみましたら、反応するようになりました。
これにて左右含めて全キー実装することができました。ありがとうございます！
